enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+---
+title: manual-cloud-credentials
+authors:
+  - "@dgoodwin"
+reviewers:
+  - @joelddiaz
+  - @smarterclayton
+  - @abhinavdahiya
+  - @derekwaynecarr
+  - @soltysh
+approvers:
+  - @joelddiaz
+  - @abhinavdahiya
+  - @soltysh
+creation-date: 2020-05-14
+last-updated: 2020-05-14
+status: provisional
+---
+
+# Support Manual Provisioning of Cloud Credentials
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal outlines a plan for fully supporting the pre-existing process for disabling the cloud-credential-operator (still running, but in a no-op mode) and manually provisioning the cloud credentials for your cluster. It allows users to avoid broken upgrades when credentials have changed, and provides them with the tools to detect and reconcile the problem.
+
+## Motivation
+
+The current best practice for installing OpenShift 4 involves providing openshift-install with `root` cloud credentials powerful enough to mint additional fine grained credentials for each component in the cluster that needs them. The root credentials are stored in a secret in the cluster. This mechanism allows each component to have only the permissions it needs, and the required permissions are defined in the release image and automatically upgraded with the cluster itself.
+
+However some customers are unable or unwilling to do this and would prefer to disable automatic credential provisioning, and instead manage it themselves. It is possible to disable the cloud-credential-operator today by injecting a `ConfigMap` into the openshift-install manifests directory, along with several `Secrets` containing the cloud provider credentials the user manually configured. While this process exists today, it is not officially supported as we believe there are situations that could arise during upgrade where credentials have changed, and the upgrade will block and go into a failed state, with little visibility as to why or how it can be fixed.
```

Comment:
```suggestion
However some customers are unable or unwilling to do this and would prefer to disable automatic credential provisioning, and instead manage it themselves. It is possible to disable the cloud-credential-operator today by injecting a `ConfigMap` into the openshift-install manifests directory, along with several `Secrets` containing the cloud provider credentials, for cluster operators, the user manually configured. While this process exists today, it is not officially supported as we believe there are situations that could arise during upgrade where credentials have changed, and the upgrade will block and go into a failed state, with little visibility as to why or how it can be fixed.
```